lib/Client.js

```diff
@@ -55,207 +76,36 @@ util.inherits(Client, Object);
-    self.getApplications({ name: name }, function(err, applications) {
-      if (err) {
-        return cb(err);
-      }
-
-      applications.detect(function(app, cb) {
-        cb(name === app.name);
-      }, function(app) {
-        if (app) {
-          applyOauthPolicy(app, cb);
-        }
-        return cb(new Error('Application not found: "' + name + '"'));
-      });
-    });
-  }
-
-  // Iterate over all account stores on the given Application, looking for all
-  // Social providers.  We'll then create a config.providers array which we'll
-  // use later on to dynamically populate all social login configurations ^^
-  function setProviders(application, cb) {
-    application.getAccountStoreMappings(function(err, mappings) {
-      if (err) {
-        return cb(err);
-      }
-
-      mappings.each(function(mapping, next) {
-        mapping.getAccountStore(function(err, accountStore) {
-          if (err) {
-            return next(err);
-          }
-
-          if (accountStore instanceof Directory) {
-            accountStore.getProvider(function(err, provider) {
-              if (err) {
-                return next(err);
-              }
-              // As long as the provider isn't Stormpath, or AD/LDAP, it must be
-              // a social directory!
-              if (provider.providerId !== 'stormpath' && provider.providerId !== 'ad' && provider.providerId !== 'ldap') {
-
-                // Remove unnecessary properties that will clutter our config!
-                delete provider.href;
-                delete provider.createdAt;
-                delete provider.updatedAt;
-
-                deepExtend(self.config.socialProviders[provider.providerId], provider, { enabled: true });
-              }
-              next();
-            });
-          } else {
-            next();
-          }
-        });
-      }, function(err) {
-        cb(err, application);
-      });
-    });
-  }
-
-  // Finds and returns an Application's default Account Store (Directory)
-  // object.  If one doesn't exist, nothing will be returned.
-  function findDefaultDirectoryHref(application, cb) {
-    application.getAccountStoreMappings(function(err, mappings) {
-      if (err) {
-        return cb(err);
-      }
-
-      mappings.detect(function(mapping, cb) {
-        cb(mapping.isDefaultAccountStore);
-      }, function(defaultMapping) {
-        if (defaultMapping) {
-          var href = defaultMapping.accountStore.href;
-
-          if (href.match(/directories/)) {
-            return cb(null, href);
-          }
-          if (href.match(/group/)) {
-            self.getGroup(href, function(err, group) {
-              return cb(err, group && group.directory.href);
-            });
-          } else {
-            return cb(null,null);
-          }
-        } else {
-          return cb(null,null);
-        }
-      });
-    });
-  }
+Client.prototype.on = function() {
+  var args = Array.prototype.slice.call(arguments);
```

Comment:
Because I think it's clearer to work against an array (and a copy) than it is to work with the [arguments](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) object. It's also easier to make errors when working against the arguments object, because most assume it's just an array, when it's not. Also, *args* is shorter and looks better to me, than having to retype *arguments*.